### PR TITLE
Update mastodon.jsx

### PR DESCRIPTION
**Pull request by diomed**
adds Croatian [I forgot before]

adds Occitan as well

### CODE REVIEW

**Comment by ykzts** [comment]
You need to add `...hr` to [line 88](https://github.com/tootsuite/mastodon/pull/2667/files#diff-3be90834b38abeceee25882fc50dbdcfR88).

**Comment by Quent-in** [comment]
What function has this file? It's lacking OC for occitan too, or maybe I haven't translated related files so far.

**Comment by diomed** [comment]
@Quent-in I've just added occitan as well. 
This is apparently needed for translation to be shown in front end interface.
so, pretty big deal.

**Comment by Quent-in** [comment]
Oh I see!
Line 60, it should be oc, not no ;)
Thank you very much for adding occitan too!

**Review by ykzts — CHANGES_REQUESTED** [review]
Can you resolve conflicts?

**Review by Gargron — APPROVED** [review]

**Comment by diomed** [comment]
@ykzts care to tell me what to fix? coz I'm not sure.

**Comment by ykzts** [comment]
@diomed I think that https://help.github.com/articles/resolving-a-merge-conflict-on-github/ will be helpful.

File: app/assets/javascripts/components/containers/mastodon.jsx [hunk 1]
```diff
@@ -50,12 +50,14 @@ import es from 'react-intl/locale-data/es';
...
 import fi from 'react-intl/locale-data/fi';
 import fr from 'react-intl/locale-data/fr';
+import hr from 'react-intl/locale-data/hr';
 import hu from 'react-intl/locale-data/hu';
 import it from 'react-intl/locale-data/it';
 import ja from 'react-intl/locale-data/ja';
 import pt from 'react-intl/locale-data/pt';
 import nl from 'react-intl/locale-data/nl';
 import no from 'react-intl/locale-data/no';
+import oc from 'react-intl/locale-data/oc';
```
Comment (ykzts) [inline app/assets/javascripts/components/containers/mastodon.jsx]: `react-intl/locale-data/oc` can not be found. React Intl seems not to supported to "oc".

Comment (diomed) [inline app/assets/javascripts/components/containers/mastodon.jsx]: but all files are there!?!

Comment (ykzts) [inline app/assets/javascripts/components/containers/mastodon.jsx]: I seen [`locale-data` directory](https://unpkg.com/react-intl@2.1.5/locale-data/) for [`react-intl`](https://github.com/yahoo/react-intl). `oc.js` does not seem to exist.

Comment (diomed) [inline app/assets/javascripts/components/containers/mastodon.jsx]: is this repo inclued somewhere as part of mastodon?
I mean - ok, lets say I include it and it doesn't exist, wouldn't it just fallback to smth?
if this is part of mastodon, can u locate it, I'm sure @Quent-in can contribute.

Comment (ykzts) [inline app/assets/javascripts/components/containers/mastodon.jsx]: This is due to external package. It is not included in this repository.

**Comment by yookoala** [comment]
@Quent-in Sadly, [react-intl](https://github.com/yahoo/react-intl) does not provide default locale data for Occitan. `'react-intl/locale-data/oc'` simply do not exists and hence cannot be imported. You'd need to implement you own locale data with their [locale data API](https://github.com/yahoo/react-intl/wiki/API#locale-data-apis).

**Comment by yookoala** [comment]
@Quent-in A simplified version of the 'en' locale data looks like this. You can create your own locale data in a similar structure. Then you may import and add with `addLocaleData`.

```javascript
[
  {
    locale: "en",
    pluralRuleFunction: function(e, a) {
      var n = String(e).split("."),
        l = !n[1],
        o = Number(n[0]) == e,
        t = o && n[0].slice(-1),
        r = o && n[0].slice(-2);
      return a ? 1 == t && 11 != r ? "one" : 2 == t && 12 != r ? "two" : 3 == t && 13 != r ? "few" : "other" : 1 == e && l ? "one" : "other"
    },
    fields: {
      year: {
        displayName: "year",
        relative: {
          0: "this year",
          1: "next year",
          "-1": "last year"
        },
        relativeTime: {
          future: {
            one: "in {0} year",
            other: "in {0} years"
          },
          past: {
            one: "{0} year ago",
            other: "{0} years ago"
          }
        }
      },
      month: {
        displayName: "month",
        relative: {
          0: "this month",
          1: "next month",
          "-1": "last month"
        },
        relativeTime: {
          future: {
            one: "in {0} month",
            other: "in {0} months"
          },
          past: {
            one: "{0} month ago",
            other: "{0} months ago"
          }
        }
      },
      day: {
        displayName: "day",
        relative: {
          0: "today",
          1: "tomorrow",
          "-1": "yesterday"
        },
        relativeTime: {
          future: {
            one: "in {0} day",
            other: "in {0} days"
          },
          past: {
            one: "{0} day ago",
            other: "{0} days ago"
          }
        }
      },
      hour: {
        displayName: "hour",
        relativeTime: {
          future: {
            one: "in {0} hour",
            other: "in {0} hours"
          },
          past: {
            one: "{0} hour ago",
            other: "{0} hours ago"
          }
        }
      },
      minute: {
        displayName: "minute",
        relativeTime: {
          future: {
            one: "in {0} minute",
            other: "in {0} minutes"
          },
          past: {
            one: "{0} minute ago",
            other: "{0} minutes ago"
          }
        }
      },
      second: {
        displayName: "second",
        relative: {
          0: "now"
        },
        relativeTime: {
          future: {
            one: "in {0} second",
            other: "in {0} seconds"
          },
          past: {
            one: "{0} second ago",
            other: "{0} seconds ago"
          }
        }
      }
    }
  }
]
```

**Comment by Quent-in** [comment]
I'll translate it as soon as possible.
Can I send it somewhere too so it will be included in other projects using this " react-intl" thing?
Best regards

**Comment by diomed** [comment]
@Gargron why was this approved, then closed?
can u reopen? if all it takes is to remove oc, I'll remove it, 
so Croatian can be merged, and then @Quent-in can make a new patch for oc

**Comment by yookoala** [comment]
@diomed: mastodon.js has been moved to another folder. The change in this PR is no longer valid. I have done the exact same change in #2844. Since it is included, this PR is consider resolved and hence closed.

**Comment by yookoala** [comment]
@Quent-in: The react-intl is quite complicated. It took me some time to figure out that source data is not in any javascript files. It is extracted from data in [CLDR - Unicode Common Locale Data Repository](http://cldr.unicode.org/). So if you really want to backport your translations, you'd need to read their [New Developers guide](http://cldr.unicode.org/development/new-cldr-developers).

It is not a lot more complicated than a Github pull request, thou.

**Comment by Quent-in** [comment]
Hello again!
I've just finished!
Here it is, I'll have a look at the link you put tonight, many thanks!

`[
  {
    locale: "oc",
    pluralRuleFunction: function(e, a) {
      var n = String(e).split("."),
        l = !n[1],
        o = Number(n[0]) == e,
        t = o && n[0].slice(-1),
        r = o && n[0].slice(-2);
      return a ? 1 == t && 11 != r ? "un" : 2 == t && 12 != r ? "dos" : 3 == t && 13 != r ? "pauc" : "autre" : 1 == e && l ? "un" : "autre"
    },
    fields: {
      year: {
        displayName: "an",
        relative: {
          0: "ongan",
          1: "l'an que ven",
          "-1": "l'an passat"
        },
        relativeTime: {
          future: {
            one: "dins {0} an",
            other: "dins {0} ans"
          },
          past: {
            one: "fa {0} an",
            other: "fa {0} ans"
          }
        }
      },
      month: {
        displayName: "mes",
        relative: {
          0: "aqueste mes",
          1: "lo mes que ven",
          "-1": "lo mes passat"
        },
        relativeTime: {
          future: {
            one: "dins {0} mes",
            other: "dins {0} meses"
          },
          past: {
            one: "fa {0} mes",
            other: "fa {0} meses"
          }
        }
      },
      day: {
        displayName: "jorn",
        relative: {
          0: "uèi",
          1: "deman",
          "-1": "ièr"
        },
        relativeTime: {
          future: {
            one: "dins {0} jorn",
            other: "dins {0} jorns"
          },
          past: {
            one: "fa {0} jorn",
            other: "fa {0} jorns"
          }
        }
      },
      hour: {
        displayName: "ora",
        relativeTime: {
          future: {
            one: "dins {0} ora",
            other: "dins {0} oras"
          },
          past: {
            one: "fa {0} ora",
            other: "fa {0} oras"
          }
        }
      },
      minute: {
        displayName: "minuta",
        relativeTime: {
          future: {
            one: "dins {0} minuta",
            other: "dins {0} minutas"
          },
          past: {
            one: "fa {0} minuta",
            other: "fa {0} minutas"
          }
        }
      },
      second: {
        displayName: "segonda",
        relative: {
          0: "ara"
        },
        relativeTime: {
          future: {
            one: "dins {0} segonda",
            other: "dins {0} segondas"
          },
          past: {
            one: "fa {0} segonda",
            other: "fa {0} segondas"
          }
        }
      }
    }
  }
]`

**Comment by yookoala** [comment]
@Quent-in: I'm working on a PR with your translations. Stay tuned.

**Comment by Quent-in** [comment]
Many thanks @yookoala !
I'm proud to spread my language, thank you for helping me.

**Comment by yookoala** [comment]
@Quent-in: Upon re-reading your translation, I think the locale data could be wrongly done.

Please take a look and the [pluralRuleFunction here](https://github.com/tootsuite/mastodon/blob/e4b84c7ba5bc4fa92f97aa8853d07931e460f87a/app/javascript/mastodon/locales/locale-data/oc.js#L6,L13). I think the usage of the pluralRuleFunction is to provide the correct key to search for plural translation.

If the field to translate is "year" and the plural function returns "one", then the program will look for "one" in the translations that structure like this:
```
future: {
     one: "dins {0} an",
     other: "dins {0} ans"
},
```

The purpose is to find the correct plural form for a given noun. If it were English, a singular item ("one") should have a string without "s", while a plural item ("other") should have a string with "s". In Chinese, there is no separated plural form so the function always returns "other".

Now. Back to your case.

Your plural function seems to return locale strings instead of "one" or "other". But your translations are still using "one", "other" and key. It could run into problem.

My advice:
1. If your language has no plural form, the plural function should just return "other".
2. If your language has 1 plural form, the plural function should return "one" or "other. No need to translate.
3. If your language has more than 1 plural form, you should have a different key for each plural form cases (preferable in English noun like "one", "two", "three", "four"). And you should have the exact same set of keys in all your translations below.

If you're not sure about git, you can give me an update version of the JS here. I can commit them for you :-)

**Pull request closed without merging**